src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
```

Comment:
I would add whole API although it doesn't affect consumption, just to be consistent with implementation, until skate is migrated to TS 
```ts
class Component extends HTMLElement {
  static readonly props: { [name: string]: PropOptions };
  static readonly observedAttributes: string[];
// ....
}
```